src/lib/AST/ASTVisitor.cpp

```diff
@@ -1723,6 +1723,11 @@ class ASTVisitor
                             return true;
                         continue;
                     }
+                    // if the class inherits from itself, we can't determine whether
+                    // it's a SFINAE type
+                    if(declaresSameEntity(TD, sfinae_info->Template))
+                      return true;
```

Comment:
is this the cause of the reported issue?

Reply:
@vinniefalco yes